File browser with shortcuts

pconesa opened the pull request:
and some usability improvements (search, keyboard navigation)

pconesa commented:
@delarosatrevin , it would be nice if you could have a look.

delarosatrevin commented:
@pconesa, I will take a look.

delarosatrevin commented on pyworkflow/gui/tree.py:
```diff
@@ -224,7 +268,7 @@ def __init__(self, master, provider, frame=True, **opts):
         self.bind('<FocusOut>', self._unpostMenu)
-        self.bind("<Key>", self._unpostMenu)
+        self.bind("<Key>", self._onKeyPress)
         self.bind('<Button-1>', self._onClick())
```
This is not introduced in the PR, but here I think should be self._onClick instead of self._onClick()

delarosatrevin commented:
Moreover, there is a problem when using the keybindings, there is always the following print:
Can't search item in browser. Using char p Item "pyworkflow" not found
Can't search item in browser. Using char c Item "config" not found
Can't search item in browser. Using char n Item "new-proj" not found

Although it properly moves to the folder starting with the given letter.
I would suggest that a lowercase letter could also match uppercase folder names...it is tedious to press also the Shift key...I think it is the default behavior of normal browsers. And an extra plus (I know that it might be complicated to implement) is to allow two or more keystrokes for the keybiding search. 

Regarding the current shortcurts:
* I find the the "root" one not much useful since now we can just type / in the path entry and type enter, so it does not add much value. 
* Every time the browser is fired from the project, the "Project dir" and the "Working dir" are the same, so I would suggest to remove one of them. 
I general I like more simplicity and adding a bunch of shortcuts does not help in all cases.

pconesa commented on pyworkflow/utils/path.py:
```diff
@@ -37,6 +37,8 @@
 import datetime
 
+ROOT = "/"
+PATH_SEP = "/"
```
ignorance? ;-)

pconesa commented on pyworkflow/utils/path.py:
```diff
@@ -86,6 +88,11 @@ def findRootFrom(referenceFile, searchFile):
+def getParentFolder(path):
+    """ Returns the parent folder of a file or folder. Work for folders that ens with "/"
+    which dirname can't"""
+    return os.path.abspath(os.path.join(path, os.pardir))
```
I think dirname remove the last /, and for this we want to keep it. I tried dirname, but it didn't work.

delarosatrevin replied:
abspath also remove the last /
so what you want to do here is to get the parent folder and you don't want to get it empty for path ending in / 
the variant that I proposed do this and does not add more characters ( /..), although it is not a big deal...except if you are in a loop and call this function many times and star piling up '/../../../../../ etc'

pconesa replied:
Ok, as usual...your solution works as well. Changed.

delarosatrevin commented:
@pconesa Have you look at the error when typing in the browser?

pconesa commented:
Nop, sorry , didn't have time.

pconesa commented:
Hi, @delarosatrevin , I was implementing the "two or more keystrokes". Let's say you have 3 folders/files named dame, dime, dome.
We need a elapsed time, let's say 1 second, after which the second type will be considered as a fresh new type, so:

1.-type "d"
2.- wait more than 1 sec
3.-type "o"

this will search for files or folder starting with "o" and not "do"

Otherwise, if you:
1.-type "d"
2- type "o" (in less than 1 second)

it will search for "do" (concatenating the 2 keystrokes).

For me, I was used to type "d"...and then "d" to move to the next folder starting with "d"...the thing is now the search tries "dd" if I do it in less than a second..
Seems we need to choose one or the other. 
Reducing the elapsed time from 1 sec to less might improve it a bit the usability...

I don't want to go into a thread like implementation. ;-(

delarosatrevin commented:
Hi @pconesa  I just suggested the two keystrokes as a wish to have...but it is not really a very important thing. I would say that still with a single keystroke is better than how it was before. After fixing the error that is display in the console, I would only suggest to remove the "Project" icon, since it is the same as "Working dir" and we can merge this PR and finally the release-1.1.1 into release-1.1.facilities-devel.

pconesa commented:
@delarosatrevin, I've added the "typed" search since it was already implemented, but only is the elapsed time is less than 0.3 seconds..otherwise it will be considered a new search. I also added a "Launch folder" shortcut and removed the Root and working folder as you suggested. The launch folder will help DLS, I hope.
